core/api/src/app/accounts/get-wallet-from-account.ts

```diff
@@ -0,0 +1,23 @@
+import { CouldNotFindWalletFromUsernameAndCurrencyError } from "@/domain/errors"
+import { WalletsRepository } from "@/services/mongoose"
+
+export const getWalletFromAccount = async (
+  account: Account,
+  walletCurrency?: WalletCurrency,
+  value?: Username | PhoneNumber,
+): Promise<Wallet | RepositoryError> => {
+  const wallets = await WalletsRepository().listByAccountId(account.id)
+  if (wallets instanceof Error) return wallets
+
+  if (!walletCurrency) {
+    return (
+      wallets.find((wallet) => wallet.id === account.defaultWalletId) ??
+      new CouldNotFindWalletFromUsernameAndCurrencyError(value)
```

Comment:
if you split this query you need to use proper errors... `CouldNotFindDefaultWalletForAccount` (which should be critical)